x-pack/plugins/fleet/server/saved_objects/migrations/security_solution/to_v8_10_0.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectUnsanitizedDoc } from '@kbn/core/server';
+
+import type { SavedObjectModelDataBackfillFn } from '@kbn/core-saved-objects-server';
+
+import type { PackagePolicy } from '../../../../common';
+
+export const migratePackagePolicyToV8100: SavedObjectModelDataBackfillFn<
+  PackagePolicy,
+  PackagePolicy
+> = (packagePolicyDoc) => {
+  if (packagePolicyDoc.attributes.package?.name !== 'endpoint') {
+    return packagePolicyDoc;
+  }
+
+  const updatedPackagePolicyDoc: SavedObjectUnsanitizedDoc<PackagePolicy> = packagePolicyDoc;
+
+  const input = updatedPackagePolicyDoc.attributes.inputs[0];
+
+  if (input && input.config) {
+    const policy = input.config.policy.value;
+
+    policy.windows.behavior_protection.reputation_service = false;
+    policy.mac.behavior_protection.reputation_service = false;
+    policy.linux.behavior_protection.reputation_service = false;
+  }
+
+  return updatedPackagePolicyDoc;
```

Comment:
Beware, a `SavedObjectModelDataBackfillFn` receives the whole document as input, but is supposed to return only the added attributes, not the whole document.

https://github.com/elastic/kibana/blob/52eb488692ca74cdc388188100540cc39e5d887e/packages/core/saved-objects/core-saved-objects-server/src/model_version/transformations.ts#L88-L90

To avoid polluting the return value, could you change

```ts
return updatedPackagePolicyDoc;
```

to 

```ts
return { attributes: updatedPackagePolicyDoc.attributes };
```

Or even better, to properly reflects which properties the function is working on

```ts
return { 
  attributes: {
    inputs:  updatedPackagePolicyDoc.attributes.inputs,
  } 
};
```

Reply:
@pgayvallet thanks for the info.

To clarify - we have three fields that we need to add in this migration with default values.

```
policy.windows.behavior_protection.reputation_service = false;
policy.mac.behavior_protection.reputation_service = false;
policy.linux.behavior_protection.reputation_service = false;
```

This `SavedObjectModelDataBackfillFn` would effectively replace the migration functions that we've provided in all previous releases, correct?  So instead of adding the migration function to migrations array as we did here: https://github.com/elastic/kibana/pull/161617/files#diff-4e164e3802d5171bf96a2cf9c91c20e97c5e0b74b2f93187a072d9a3139f1c18R299 - we use this model update function.

thanks for your help on this!

Reply:
@pgayvallet 
Thanks for your help, Pierre! I implemented the changes that you requested. Could you take a look whenever you find time? 1c67e42c2b4953003bc0fd2631f5929a3b14578e